### PR TITLE
*: enable `asm` feature for the `sha2` crate

### PR DESCRIPTION
**Pull request by petrosagg**
### Motivation

Recent benchmarking revealed that we spend a significant amount of CPU time computing SHA256 hashes which can be accelerated a lot by using assembly implementations that can take advantage of specialized CPU instructions.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

**Comment by petrosagg** [comment]
Is there a more "proper" way of enabling this feature than a random crate that happens to use the `sha2` crate? Should I add it as a dependency to one of `persist`'s crates with a comment?

**Review by benesch — APPROVED** [review]
Nice!

**Comment by benesch** [comment]
No more proper way that I’m aware of. Though I would try to put it as close to the crate that needs it as possible. Putting this in a persist crate makes sense to me. You can introduce a dependency where it didn’t previously exist. There’s precedent for that. Look for cargo udeps ignores to see what I mean.

**Comment by petrosagg** [comment]
Cool, added it as a dependency in `persist` with a comment

**Comment by benesch** [comment]
LGTM, but @MaterializeInc/persist PTAL and holler if this objectionable.

**Comment by danhhz** [comment]
LGTM

extreme nitpick: Having comments inline in Cargo.toml dependency sections is a touch annoying because it means you can't just select the block of them and sort ascending with tooling. Personally, I'd put a fake usage of the crate in mz_persist somewhere (so that it's actually used) and stick the comment on that, but it's marginal enough that I wouldn't push back if you didn't want to bother

**Review by danhhz — APPROVED** [review]

File: src/persist/Cargo.toml [hunk 1]
```diff
@@ -62,6 +62,14 @@ url = "2.3.1"
 uuid = { version = "1.7.0", features = ["v4"] }
 workspace-hack = { version = "0.0.0", path = "../workspace-hack" }
 
+
+# While this crate doesn't directly depend on `sha2` we want to ensure that the
+# `asm` feature is enabled since computing SHA256 hashes is taking a
+# significant amount of CPU time when uploading data to S3
+[dependencies.sha2]
```
Comment (danhhz) [inline src/persist/Cargo.toml:69]: This new version also works for me, thanks!

**Comment by petrosagg** [comment]
Moved the dependency into its own section so that the comment doesn't interfere

**Pull request merged**